src/snowflake/connector/config_manager.py

```diff
@@ -334,44 +362,48 @@ def add_subparser(self, new_child: ConfigManager) -> None:
             new_child: The ConfigManager to be nested under the current one.
         Notes:
             We currently don't support re-nesting a ConfigManager. Only nest a
-            parser under another one once.
+            manager under another one once.
         """
         self._check_child_conflict(new_child.name)
-        self._sub_parsers[new_child.name] = new_child
+        self._sub_managers[new_child.name] = new_child
 
         def _root_setter_helper(node: ConfigManager):
             # Deal with ConfigManagers
             node._root_manager = self._root_manager
             node._nest_path = self._nest_path + node._nest_path
-            for sub_parser in node._sub_parsers.values():
-                _root_setter_helper(sub_parser)
+            for sub_manager in node._sub_managers.values():
+                _root_setter_helper(sub_manager)
             # Deal with ConfigOptions
             for option in node._options.values():
                 option._root_manager = self._root_manager
                 option._nest_path = self._nest_path + option._nest_path
 
         _root_setter_helper(new_child)
 
+    # Alias for the old name of add_submanager in the first release, please use the
+    #  new name, as this might get deprecated in the future.
+    add_subparser = add_submanager
+
     def __getitem__(self, name: str) -> ConfigOption | ConfigManager:
-        """Get either sub-parser, or option in this parser with name.
+        """Get either sub-manager, or option in this manager with name.
 
-        If option is retrieved, we call get() on it to return its value instead.
+        If an option is retrieved, we call get() on it to return its value instead.
 
         Args:
             name: Name to retrieve.
         """
         if name in self._options:
             return self._options[name].value()
-        if name not in self._sub_parsers:
+        if name not in self._sub_managers:
             raise ConfigSourceError(
                 "No ConfigManager, or ConfigOption can be found"
                 f" with the name '{name}'"
             )
-        return self._sub_parsers[name]
+        return self._sub_managers[name]
 
 
-CONFIG_PARSER = ConfigManager(
-    name="CONFIG_PARSER",
+CONFIG_MANAGER = ConfigManager(
+    name="CONFIG_MANAGER",
```

Comment:
is `name` used any where in the code? shall we take compatibility into account when dealing with the default name value `CONFIG_PARSER` -> `CONFIG_MANAGER`

Reply:
It's only used in Exceptions we raise.

Reply:
Generally in Python, we consider exception types to be a backward compatibility promise (though it depends, and not always), but exception *messages* are not usually considered part of the API and thus not dependent on backward compatibility.